frontend/packages/console-shared/src/selectors/node.ts

```diff
@@ -1,68 +1,56 @@
-import * as _ from 'lodash';
-import { NodeKind } from '@console/internal/module/k8s';
-import { NodeAddress, NodeCondition } from '../types';
+import {
+  NodeAddress,
+  NodeKind,
+  Taint,
+} from '@console/dynamic-plugin-sdk/src/extensions/console-types';
 
 const NODE_ROLE_PREFIX = 'node-role.kubernetes.io/';
 
-export const getNodeRoles = (node: NodeKind): string[] => {
-  const labels = _.get(node, 'metadata.labels');
-  return _.reduce(
-    labels,
-    (acc: string[], v: string, k: string) => {
-      if (k.startsWith(NODE_ROLE_PREFIX)) {
-        acc.push(k.slice(NODE_ROLE_PREFIX.length));
-      }
-      return acc;
-    },
-    [],
-  );
-};
+export const getNodeRoles = (node: NodeKind): string[] =>
+  Object.keys(node?.metadata?.labels ?? {}).reduce((acc: string[], k) => {
```

Comment:
Nit: when using `Array.prototype.reduce` function, it's better to type the accumulator like so

```ts
Object.keys(node?.metadata?.labels ?? {}).reduce<string[]>((acc, k) => {
  // ...
}, []);
```